File: src/main/java/com/sip/dmesmobile/utilities/Utilities.java
package com.sip.dmesmobile.utilities;

import java.lang.reflect.Field;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


/**
 * Proyecto: DMES - Versión: 1.0
 * @author (es): Gustavo Adolfo Chavarro Ortiz 
 * @Funcionalidad: Esta clase está encargada contener aquellas funciones que 
 * nos pueden ser utilies para diferentes clases o funcionalidades de la aplicación.
 * @Modificado por:  
 * @Funcionalidad agregada:
 */
public class Utilities
{

    //Arreglo con números y letras para generar la encriptación
    private static final char[] CONSTS_HEX =
    {
        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
    };

    /**
     * Método encargado de redondear una cantidad recibida a un número de dígitos indicados
     * @author: Gustavo Chavarro Ortiz
     * @param numero valor a redondear
     * @param digitos cantidad de digitos decimales a retornar
     * @return double valor a retornar
     */
    public static double Redondear(double numero, int digitos)
    {
        int cifras = (int) Math.pow(10, digitos);
        return Math.rint(numero * cifras) / cifras;
    }

    /**
     * Método encargado de de verificar si una cadena esta vacia
     * @author: Gustavo Chavarro Ortiz
     * @param value valor de la cadena a verificar
     * @return boolean indica si esta vacio o no
     */
    public static boolean isEmpty(String value)
    {
        boolean result = true;
        if (value != null)
        {
            value = value.replace(" ", "");
            if (value.length() > 0)
            {
                result = false;
            }
        }
        return result;
    }

    /**
     * Método encargado de indicar si una cadena es totalmente numerica
     * @author: Gustavo Chavarro Ortiz
     * @param value cadena recibida a verificar
     * @return boolean inidica si es numerico o no
     */
    public static boolean isNumeric(String value)
    {
        boolean result = false;
        String regex = "[0-9]*";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(value);
        if (matcher.matches())
        {
            result = true;
        }
        return result;
    }

    /**
     * Método encargado de indicar si una cadena es alfanumerica
     * @author: Gustavo Chavarro Ortiz
     * @param value cadena recibida a verificar
     * @return boolean inidica si es alfanumerica o no
     */
    public static boolean isAlphaNumeric(String value)
    {
        boolean result = false;
        String regex = "[a-zA-Z0-9Ññ]*";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(value);
        if (matcher.matches())
        {
            result = true;
        }
        return result;
    }

    /**
     * Método encargado de indicar si una cadena es alfabetica
     * @author: Gustavo Chavarro Ortiz
     * @param value cadena recibida a verificar
     * @return boolean inidica si es alfabetica o no
     */
    public static boolean isAlphabetical(String value)
    {
        boolean result = false;
        String regex = "[a-zA-ZÑñ\\t\\n\\x0b\\r\\f]*";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(value);
        if (matcher.matches())
        {
            result = true;
        }
        return result;
    }

    /**
     * Método encargado de indicar si una cadena es un email
     * @author: Gustavo Chavarro Ortiz
     * @param value cadena recibida a verificar
     * @return boolean inidica si es un email o no
     */
    public static boolean isEmail(String value)
    {
        boolean result = false;
        String regex = "^[\\w-]+(\\.[\\w-]+)*@[A-Za-z0-9]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(value);
        if (matcher.matches())
        {
            result = true;
        }
        return result;
    }

    /**
     * Método encargado de encriptar una cadena recibida
     * @author: Gustavo Chavarro Ortiz
     * @param stringAEncriptar cadena recibida a encriptar
     * @return String valor de la candena encriptada
     */
    public static String encriptaEnMD5(String stringAEncriptar)
    {
        try
        {
            MessageDigest msgd = MessageDigest.getInstance("MD5");
            byte[] bytes = msgd.digest(stringAEncriptar.getBytes());
            StringBuilder strbCadenaMD5 = new StringBuilder(2 * bytes.length);
            for (int i = 0; i < bytes.length; i++)
            {
                int bajo = (int) (bytes[i] & 0x0f);
                int alto = (int) ((bytes[i] & 0xf0) >> 4);
                strbCadenaMD5.append(CONSTS_HEX[alto]);
                strbCadenaMD5.append(CONSTS_HEX[bajo]);
            }
            return strbCadenaMD5.toString();
        }
        catch (NoSuchAlgorithmException e)
        {
            return null;
        }
    }
    
    public static String fromArrayToJSON(Object list, String className, String nameArray)throws Exception
    {
        
        String result = "{\""+nameArray+"\":[";
        for(Object object: ((List<Object>) list))
        {
            result += fromObjectToJSON(object, className)+",";
        }
        result = result.substring(0, result.length()-1);
        result += "]}";
        return result;
    }
    
    public static String fromObjectToJSON(Object object, String className)throws Exception
    {
        String result = "{";
        if(object != null && className != null && className.length() > 0)
        {
                Class ObjectClass = Class.forName(className);
                Field[] fields = ObjectClass.getDeclaredFields();
                for(int i=0; i< fields.length; i++)
                {
                    if(isNative(fields[i].getGenericType().toString()))
                    {
                        if(!fields[i].getName().equals("serialVersionUID"))
                        {
                            result+= "\""+fields[i].getName()+"\":\""+fields[i].get(object)+"\"";
                            result+=","; 
                        }
                    }
                    else if(isForeing(fields[i].getGenericType().toString()))
                    {
                        if(!isList(fields[i].getGenericType().toString()))
                        {
                            result+= "\""+fields[i].getName()+"\":";
                            result+= fromObjectToJSON(fields[i].get(object), fields[i].getGenericType().toString().replace("class ", ""));
                            result+=",";
                        }
                    }
                    else if(isList(fields[i].getGenericType().toString()))
                    {
                        result+= "\""+fields[i].getName()+"\":";
                        result+= fromArrayToJSON(fields[i].get(object), ((List<Object>) fields[i].get(object)).get(0).getClass().getCanonicalName(), fields[i].getName());
                        result+=",";
                    }
                } 
                result = result.substring(0, result.length()-1);
            
        }
        result += "}";
        return result;
    }
    
    public static boolean isNative(String type)
    {
        boolean result = true;
        
        if(type.contains("co.sip.dmesmobile.entitys") || type.contains("java.util.List"))
        {
            result = false;
        }
        return result;
    }
    
    public static boolean isList(String type)
    {
        boolean result = false;
        if(type.contains("java.util.List"))
        {
            result = true;
        }
        return result;
    }
    
    public static boolean isForeing(String type)
    {
        boolean result = false;
        
        if(type.contains("co.sip.dmesmobile.entitys"))
        {
            result = true;
        }
        return result;
    }

    public static String messageJSON(String message)
    {
        return "{\"message\":\""+message+"\"}";
    }
    
    
    
    
}

File: src/main/webapp/frontEnd/utilities/utilities.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */




/**
 * Método encargado de redondear una cantidad recibida a un número de dígitos indicados
 * @author: Gustavo Chavarro Ortiz
 * @param numero valor a redondear
 * @param digitos cantidad de digitos decimales a retornar
 * @return double valor a retornar
 */
function redondear(numero, digitos)
{
    var cifras = 1;
    var position = 0;
    while (position < digitos)
    {
        cifras = 10 * cifras;
        position++;
    }
    var flotante = parseFloat(numero);
    var resultado = Math.round(flotante * cifras) / cifras;
    return resultado;
}

/**
 * Método encargado de de verificar si una cadena esta vacia
 * @author: Gustavo Chavarro Ortiz
 * @param value valor de la cadena a verificar
 * @return boolean indica si esta vacio o no
 */
function isEmpty(value)
{
    var result = true;
    if (value !== null)
    {
        value = value.replace(/ /g, '');
        if (value.length > 0)
        {
            result = false;
        }
    }
    return result;
}

/**
 * Método encargado de indicar si una cadena es totalmente numerica
 * @author: Gustavo Chavarro Ortiz
 * @param value cadena recibida a verificar
 * @return boolean inidica si es numerico o no
 */
function isNumeric(value)
{
    var result = false;
    if (/^[0-9]*$/.test(value))
    {
        result = true;
    }
    return result;
}


/**
 * Método encargado de indicar si una cadena es un email
 * @author: Gustavo Chavarro Ortiz
 * @param value cadena recibida a verificar
 * @return boolean inidica si es un email o no
 */
function isEmail(value)
{
    var result = false;
    if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,4})+$/.test(value))
    {
        result = true;
    }
    return result;
}



function convertStringToJSON(value)
{
    value = value.replace(/\n/g, '');
    value = JSON.parse(value);
    return value;
}

function addInfoMessage(title, message, time)
{
    time = time * 1000;
    $.unblockUI();
    $("#titleMessage").html("</br>" + title + "</br>");
    $("#contentMessage").html("</br>" + message + "</br></br>");
    $("#messages").popup("open");
    setTimeout(function()
    {
        $("#messages").popup("close");
    }, 5000);

}

function setCookie(cname, cvalue, years) 
{
    var d = new Date();
    d.setTime(d.getTime() + (years * 360 * 24 * 60 * 60 * 1000));
    var expires = "expires=" + d.toGMTString();
    document.cookie = cname + "=" + cvalue + "; " + expires;
}

function getCookie(cname) 
{
    var name = cname + "=";
    var ca = document.cookie.split(';');
    for (var i = 0; i < ca.length; i++) {
        var c = ca[i];
        while (c.charAt(0) == ' ')
            c = c.substring(1);
        if (c.indexOf(name) == 0) {
            return c.substring(name.length, c.length);
        }
    }
    return "";
}

function checkCookie(nameCookie) 
{
    var cookie = getCookie(nameCookie);
    if (cookie != "") 
    {
        addInfoMessage("Valor Registrado", "El valor de "+nameCookie+" = "+cookie+" ", 5);
        MACHINE_SELECTED = cookie;
        return true;
    } else 
    {
        return false;
    }
}

function getOrderNumber(idValue, creationDate, prefix)
{
    creationDate = creationDate.replace(/ /g, "");
    creationDate = creationDate.replace(/-/g, "");
    creationDate = creationDate.replace(/:/g, "");
    var position = creationDate.indexOf('.');
    creationDate = creationDate.substring(0, position-1);
    return prefix+creationDate+idValue;
}

function getDate(creationDate)
{
    creationDate = creationDate.substring(0, creationDate.length-2);
    return creationDate;
}


function removeAccent(value)
{
    value = value.replace(/á/g, "a");
    value = value.replace(/é/g, "e");
    value = value.replace(/í/g, "i");
    value = value.replace(/ó/g, "o");
    value = value.replace(/ú/g, "u");
    value = value.replace(/Á/g, "A");
    value = value.replace(/É/g, "E");
    value = value.replace(/Í/g, "I");
    value = value.replace(/Ó/g, "O");
    value = value.replace(/Ú/g, "U");
    value = value.replace(/ñ/g, "n");
    value = value.replace(/Ñ/g, "N");
    return value;
}
